x/poe/simulation/operations.go

```diff
@@ -0,0 +1,339 @@
+			Cdc:           nil,
+			Msg:           msg,
+			MsgType:       msg.Type(),
+			Context:       ctx,
+			SimAccount:    simAccount,
+			ModuleName:    types.ModuleName,
+		}
+
+		return simulation.GenAndDeliverTx(txCtx, fees)
+	}
+}
+
+// SimulateMsgUpdateValidator generates a MsgUpdateValidator with random values
+func SimulateMsgUpdateValidator(bk types.XBankKeeper, ak types.AccountKeeper, k keeper.Keeper) simtypes.Operation {
+	return func(
+		r *rand.Rand, app *baseapp.BaseApp, ctx sdk.Context, accs []simtypes.Account, chainID string,
+	) (simtypes.OperationMsg, []simtypes.FutureOperation, error) {
+		validators, _, err := k.ValsetContract(ctx).ListValidators(ctx, nil)
+		if len(validators) == 0 || err != nil {
+			return simtypes.NoOpMsg(types.ModuleName, types.TypeMsgUpdateValidator, "numbers of validator equal zero"), nil, nil
+		}
+
+		val := validators[rand.Intn(len(validators))]
+		accAddr, err := sdk.AccAddressFromBech32(val.OperatorAddress) // plain string to not run into bech32 prefix issues with valoper
+		if err != nil {
+			return simtypes.NoOpMsg(types.ModuleName, types.TypeMsgUpdateValidator, "validator address is not valid"), nil, err
+		}
+
+		simAccount, found := simtypes.FindAccount(accs, accAddr)
+		if !found {
+			return simtypes.NoOpMsg(types.ModuleName, types.TypeMsgUpdateValidator, "unable to find account"), nil, fmt.Errorf("validator %s not found", val.GetOperator())
+		}
+
+		spendable := bk.SpendableCoins(ctx, simAccount.Address)
+
+		description := stakingtypes.NewDescription(
+			simtypes.RandStringOfLength(r, 10),
+			simtypes.RandStringOfLength(r, 10),
+			"https://"+simtypes.RandStringOfLength(r, 10),
+			simtypes.RandStringOfLength(r, 10),
+			simtypes.RandStringOfLength(r, 10),
+		)
+
+		msg := types.NewMsgUpdateValidator(accAddr, description)
+
+		txCtx := simulation.OperationInput{
+			R:               r,
+			App:             app,
+			TxGen:           simappparams.MakeTestEncodingConfig().TxConfig,
+			Cdc:             nil,
+			Msg:             msg,
+			MsgType:         msg.Type(),
+			Context:         ctx,
+			SimAccount:      simAccount,
+			AccountKeeper:   ak,
+			Bankkeeper:      bk,
+			ModuleName:      types.ModuleName,
+			CoinsSpentInMsg: spendable,
+		}
+
+		return simulation.GenAndDeliverTxWithRandFees(txCtx)
+	}
+}
+
+// SimulateMsgDelegate generates a MsgDelegate with random values
+func SimulateMsgDelegate(bk types.XBankKeeper, ak types.AccountKeeper, k keeper.Keeper) simtypes.Operation {
+	return func(
+		r *rand.Rand, app *baseapp.BaseApp, ctx sdk.Context, accs []simtypes.Account, chainID string,
+	) (simtypes.OperationMsg, []simtypes.FutureOperation, error) {
+
+		validators, _, err := k.ValsetContract(ctx).ListValidators(ctx, nil)
+		if len(validators) == 0 || err != nil {
+			return simtypes.NoOpMsg(types.ModuleName, types.TypeMsgDelegate, "numbers of validator equal zero"), nil, nil
+		}
+
+		simAccount, _ := simtypes.RandomAcc(r, accs)
```

Comment:
Unlike PoS only operators can delegate. So instead you would select a random validator and delegate from their validator operator address. The code would same as L160-176 and then delegate the spendable amount